code/game/objects/effects/misc.dm

```diff
@@ -71,6 +71,14 @@
 	if(icon_state == "blocker")
 		icon_state = ""
 
+/obj/effect/forcefield/dense0forguns
+	resistance_flags = PROJECTILE_IMMUNE
```

Comment:
what is this for? the parent can block projectiles if that's what you want

Reply:
It makes projectiles pass through it